sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -223,11 +224,18 @@ class Dataset[T] private[sql](
   @transient private[sql] val logicalPlan: LogicalPlan = {
     // For various commands (like DDL) and queries with side effects, we force query execution
     // to happen right away to let these side effects take place eagerly.
+    def eagerRun(plan: LogicalPlan): LogicalPlan = {
+      val relation =
+        LocalRelation(plan.output, withAction("command", queryExecution)(_.executeCollect()))
```

Comment:
The problem is we do submit two Spark jobs (invoking the command and collecting the `LocalRelation`), and it's unclear which one we should wrap with a SQL execution.

Reply:
Yes. 
IIUC, For showing SQL queries and associated jobs in Spark UI, only tracking the SQL execution is enough. 
But in general, I think it's better to track all the execution ids and let the event listeners decide the behavior for the different events for one SQL query.
So another option may be that we still track and send all the SQL execution-related events to listeners, and handle the first event of a SQL query in UI listeners.